arch/generic/compare256_c.c

```diff
@@ -8,59 +8,8 @@
 #include "deflate.h"
 #include "fallback_builtins.h"
 
-/* ALIGNED, byte comparison */
-static inline uint32_t compare256_c_static(const uint8_t *src0, const uint8_t *src1) {
-    uint32_t len = 0;
-
-    do {
-        if (*src0 != *src1)
-            return len;
-        src0 += 1, src1 += 1, len += 1;
-        if (*src0 != *src1)
-            return len;
-        src0 += 1, src1 += 1, len += 1;
-        if (*src0 != *src1)
-            return len;
-        src0 += 1, src1 += 1, len += 1;
-        if (*src0 != *src1)
-            return len;
-        src0 += 1, src1 += 1, len += 1;
-        if (*src0 != *src1)
-            return len;
-        src0 += 1, src1 += 1, len += 1;
-        if (*src0 != *src1)
-            return len;
-        src0 += 1, src1 += 1, len += 1;
-        if (*src0 != *src1)
-            return len;
-        src0 += 1, src1 += 1, len += 1;
-        if (*src0 != *src1)
-            return len;
-        src0 += 1, src1 += 1, len += 1;
-    } while (len < 256);
-
-    return 256;
-}
-
-Z_INTERNAL uint32_t compare256_c(const uint8_t *src0, const uint8_t *src1) {
-    return compare256_c_static(src0, src1);
-}
-
-#define LONGEST_MATCH       longest_match_c
-#define COMPARE256          compare256_c_static
-
-#include "match_tpl.h"
-
-#define LONGEST_MATCH_SLOW
-#define LONGEST_MATCH       longest_match_slow_c
-#define COMPARE256          compare256_c_static
-
-#include "match_tpl.h"
-
-#if OPTIMAL_CMP >= 32
-
-/* 16-bit unaligned integer comparison */
-static inline uint32_t compare256_unaligned_16_static(const uint8_t *src0, const uint8_t *src1) {
+/* 16-bit integer comparison */
+static inline uint32_t compare256_16_static(const uint8_t *src0, const uint8_t *src1) {
```

Comment:
It might be better to keep the byte version rather than the 16-bit version - the byte version would potentially have better performance than the 16-bit version on platforms without unaligned reads, and platforms without dedicated clz/ctz instructions would likely end up using the 32-bit or 64-bit versions anyway because GCC typically provides generic fallbacks for `__builtin_clz`/`__builtin_ctz` instructions for all architectures.

Reply:
I think I would have to agree, that keeping the c versions is probably a good idea for those platforms.

Reply:
Added the 8-bit version back, but not currently in use by any platform.

Reply:
The code you have now is a good compromise.